packages/contracts-bedrock/src/libraries/Slot.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/// @title Slot
+/// @notice Slot handles reading and writing to arbitary storage locations
+library Slot {
+    /// @notice Returns an address stored in an arbitrary storage slot.
+    ///         These storage slots decouple the storage layout from
+    ///         solc's automation.
+    /// @param _slot The storage slot to retrieve the address from.
+    function getAddress(bytes32 _slot) internal view returns (address addr_) {
+        assembly {
+            addr_ := sload(_slot)
+        }
+    }
+
+    /// @notice Stores an address in an arbitrary storage slot, `_slot`.
+    /// @param _slot The storage slot to store the address in.
+    /// @param _address The protocol version to store
+    /// @dev WARNING! This function must be used cautiously, as it allows for overwriting addresses
+    ///      in arbitrary storage slots.
+    function setAddress(bytes32 _slot, address _address) internal {
+        assembly {
+            sstore(_slot, _address)
+        }
+    }
+
+    /// @notice Returns a uint256 stored in an arbitrary storage slot.
+    ///         These storage slots decouple the storage layout from
+    ///         solc's automation.
+    /// @param _slot The storage slot to retrieve the address from.
+    function getUint(bytes32 _slot) internal view returns (uint256 value_) {
+        assembly {
+            value_ := sload(_slot)
+        }
+    }
+
+    /// @notice Stores a value in an arbitrary storage slot, `_slot`.
```

Comment:
Ideally we can get the changes to this contract in https://github.com/ethereum-optimism/optimism/pull/7428

Reply:
Those changes won't be on this branch.
If 7428 does get merged in, will this PR be blocked until the feature branch is rebased?